packages/core/core/src/loadEnv.js

```diff
@@ -1,23 +1,27 @@
+// @flow strict-local
+
 import {resolveConfig} from '@parcel/utils/src/config';
 import dotenv from 'dotenv';
 import variableExpansion from 'dotenv-expand';
 
-export default async function loadEnv(filepath) {
-  const NODE_ENV = process.env.NODE_ENV || 'development';
+export default async function loadEnv(filePath: string): Promise<void> {
+  const NODE_ENV =
+    process.env.NODE_ENV == null ? 'development' : process.env.NODE_ENV;
+
   const dotenvFiles = [
     `.env.${NODE_ENV}.local`,
     `.env.${NODE_ENV}`,
     // Don't include `.env.local` for `test` environment
     // since normally you expect tests to produce the same
     // results for everyone
-    NODE_ENV !== 'test' && '.env.local',
+    NODE_ENV === 'test' ? null : '.env.local',
     '.env'
   ].filter(Boolean);
 
   await Promise.all(
     dotenvFiles.map(async dotenvFile => {
-      const envPath = await resolveConfig(filepath, [dotenvFile]);
-      if (envPath) {
+      const envPath = await resolveConfig(filePath, [dotenvFile]);
+      if (envPath != null) {
```

Comment:
avoiding a sketchy null check. Behavior change: we no longer take this path if `envPath` is an empty string.